Assets/Scripts/Control&Input/PlayerMovement.cs

```diff
@@ -243,59 +277,68 @@ private void OnCrouch(InputAction.CallbackContext ctx)
 
         if (ctx.performed)
         {
-            if (IsInAir())
-            {
-                onLanding += StartCrouch;
-                return;
-            }
-            StartCrouch();
+            StartCrouching();
         }
 
         if (ctx.canceled)
         {
-            animator.SetBool("Crouching", false);
-            strafeForce = strafeForceGrounded;
-            cameraCrouchCanceledTween = inputManager.gameObject.LeanMoveLocalY(localCameraHeight, 0.2f).id;
-            gunCrouchCanceledTween = gunHolder.LeanMoveLocalY(localGunHolderHeight, 0.2f).id;
-            isDashing = false;
-            onLanding -= StartCrouch;
+            StopCrouching();
         }
-
     }
 
-    protected virtual void StartCrouch()
+    private void StartCrouching()
     {
+        isCrouching = true;
+        if (state is GroundState.InAir)
+            return;
         animator.SetBool("Crouching", true);
         strafeForce = strafeForceCrouched;
         cameraCrouchPerformedTween = inputManager.gameObject.LeanMoveLocalY(localCameraHeight - crouchedHeightOffset, 0.2f).id;
         gunCrouchCanceledTween = gunHolder.LeanMoveLocalY(localGunHolderHeight - crouchedHeightGunOffset, 0.2f).id;
     }
 
-    private void EnableDash()
+    private void StopCrouching()
     {
-        StartCoroutine(JumpTimeout(leapTimeout));
+        isCrouching = false;
+        if (state is GroundState.Grounded)
```

Comment:
I would like us not to do this if check. There is a little known feature I like a lot where if you cancel your crouch mid air then hold crouch again before you land to perform a high jump. (leaphop-cancelling basicly) I really enjoy the control this grants in some occasions (example below), but removing "leaphop-cancelling" is of course also an option if we decide we want to.

Here is an example where I removed the if check (leap jumping, letting go of crouch and then holding it again while in the same jump before perfoming a new jump):
![CrouchJumpwithout](https://github.com/hackerspace-ntnu/Red-Planet-Rampage/assets/54811121/ebeb208f-58f1-4daa-bc90-4df7c7ff2310)
And one as is here (same inputs):
![CrouchJumpwith](https://github.com/hackerspace-ntnu/Red-Planet-Rampage/assets/54811121/d755620d-c706-4858-9026-55a40eacac63)

Reply:
Think I got the intention here so I merge this now >:)